toolbox

```diff
@@ -685,6 +689,7 @@ create()
             --volume /media:/media:rslave \
             --volume /mnt:/mnt:rslave \
             --volume /run/media:/run/media:rslave \
+            ${TOOLBOX_CREATE_ARGS:-} \
```

Comment:
Yes, this has been requested before. I wonder if we should use a command line flag instead of an environment variable - in case we rewrite the script as a proper Go frontend to `libpod`.

Reply:
Also on this topic I don't agree a CLI flag is right (and either was my original change) - this stuff should "Just Work" really.

Reply:
Yes, I agree that it should just work. What else do you think should be bind mounted in from the host?

Reply:
> Also on this topic I don't agree a CLI flag is right (and either was my original change) - this stuff should "Just Work" really.

FWIW, I'm interested in this issue because I had to bind-mount some crazy stuff to get a toolbox that allows me to run gnome-shell/mutter from jhbuild.

That's a very specialized use case, so I don't think those mounts are generally useful.

Reply:
What kind of things did you need?

Reply:
/run/systemd, /run/udev, /dev/input and (for some reason) /tmp.

I also threw in /var/lib/flatpak to keep my apps in the jhbuild environment.

There may be some point in exposing the latter by default, but the mutter/shell specific bits are really rather specific.

Reply:
I filed https://github.com/debarshiray/toolbox/pull/256 for the `/var/lib/flatpak` bind mount.

`/dev/input` should already be available for any containers with a non-ancient Toolbox because we have been exposing the entire /dev for the past few releases.

Why did you need `/run/systemd` and `/run/udev`?

Reply:
I don't remember the exact details/errors, but neither systemd nor udev were terribly surprising considering that running mutter/gnome-shell as wayland compositor requires logind (session/seat management) and udev (input device hotplug).

I may have some time later this week to recreate the toolbox from scratch, to check what customizations are still required.